src/visualization/parsers/boxmodel.ts

```diff
@@ -0,0 +1,186 @@
+/// <reference path="../../../node_modules/clarity-js/clarity.d.ts" />
+import { IParser } from "../components/Snapshot";
+
+export default class BoxModel implements IParser {
+    private layouts: { [index: number]: Node } = {};
+    private states: { [index: number]: IElementLayoutState } = {};
+    private document: Document;
+    private base: string;
+   
+    setup(document: Document, frame: HTMLIFrameElement, base: string, thumbnail? : boolean) {
+        this.layouts = {};
+        this.document = document;
+        this.base = base;
+    }
+
+    private domInsert(node: Node, parent?: Node, nextSibling?: Node) {
+        if (parent) {
+            if (nextSibling && nextSibling.parentNode === parent) {
+                nextSibling.parentNode.insertBefore(node, nextSibling);
```

Comment:
>insertBefore [](start = 39, length = 12)

I think we should have a try-catch that includes this as well as the appendChild on 23

Reply:
Done, thanks.